server/src/test/java/io/crate/expression/scalar/SubscriptFunctionTest.java

```diff
@@ -43,6 +43,28 @@ public void test_long_can_be_used_as_array_index() {
         assertEvaluate("['Youri', 'Ruben'][x]", "Youri", Literal.of(1L));
     }
 
+    @Test
+    public void test_index_out_of_range_array_access() {
+        assertThatThrownBy(
+            () -> assertEvaluate("['Youri', 'Ruben'][x]",
+                                 null,
+                                 Literal.of((long) Integer.MAX_VALUE + 1)))
+            .isExactlyInstanceOf(ConversionException.class)
+            .hasMessage("Cannot cast value `2147483648` to type `integer`");
+        assertThatThrownBy(
+            () -> assertEvaluate("['Youri', 'Ruben'][x]",
+                                 null,
+                                 Literal.of((long) Integer.MIN_VALUE - 1)))
+            .isExactlyInstanceOf(ConversionException.class)
+            .hasMessage("Cannot cast value `-2147483649` to type `integer`");
+    }
+
+    @Test
+    public void test_valid_min_and_max_array_index_access() {
+        assertNormalize("subscript([], 2147483647)", isLiteral(null));
+        assertNormalize("subscript([], -2147483648)", isLiteral(null));
```

Comment:
I don't think this test 100% covers the `null` return for 0 and negative values.
Please use an array with some elements instead.